symengine/functions.cpp

```diff
@@ -2647,6 +2661,29 @@ std::size_t LeviCivita::__hash__() const
     return seed;
 }
 
+RCP<const Basic> LeviCivita::subs(const map_basic_basic &subs_dict) const
+{
+    auto it = subs_dict.find(rcp_from_this());
+    if (it != subs_dict.end())
+        return it->second;
+    vec_basic arg(arg_.size());
+    bool this_rcp = true;
+    auto it2 = arg.begin();
+    for (const auto& it: arg_) {
+        *it2 = ((it)->subs(subs_dict));
+        if (this_rcp) {
```

Comment:
If we don't want to use 2 iterators : 
(1) iterate over `arg_`
 `push_back` in `arg`
 compare `*it` with `*(arg.end()-1)` 
(2) just `push_back` in `arg` and then compare vectors by `==` or `std::mismatch`

however I doubt efficiency of above method over currently implemented method, as the current method stops doing `eq()` checks as soon as `1` mismatch is found.